demo/site/src/documentTypes/Page.tsx

```diff
@@ -47,29 +48,45 @@ export default async function Page({ pageTreeNodeId, scope }: { pageTreeNodeId:
     if (draftMode().isEnabled) {
         previewData = { includeInvisible: false };
     }
-    const graphqlFetch = createGraphQLFetch(previewData);
+    const graphQLFetch = createGraphQLFetch(previewData);
 
-    const props = await graphqlFetch<GQLPageQuery, GQLPageQueryVariables>(pageQuery, {
+    const props = await graphQLFetch<GQLPageQuery, GQLPageQueryVariables>(pageQuery, {
         pageTreeNodeId,
         domain: scope.domain,
         language: scope.language,
     });
 
     if (!props.pageContent) throw new Error("Could not load page content");
-    const document = props.pageContent.document;
-    if (!document) {
+    if (!props.pageContent.document) {
         // no document attached to page
         notFound(); //no return needed
     }
-    if (document.__typename != "Page") throw new Error(`invalid document type, expected Page, got ${document.__typename}`);
+    if (props.pageContent.document?.__typename != "Page") throw new Error(`invalid document type`);
+
+    [props.pageContent.document.content, props.pageContent.document.seo] = await Promise.all([
```

Comment:
I recently learned that [mutating props is an anti-pattern](https://react.dev/reference/rules/components-and-hooks-must-be-pure#props), so I think we shouldn't do it in Demo, even if it probably will never be a problem here.

Reply:
This variable doesn't contain props, it's named confusing. I renamed it to data.